ckanext/datastore/db.py

```diff
@@ -430,7 +430,8 @@ def create_indexes(context, data_dict):
 
     if primary_key is not None:
         _drop_indexes(context, data_dict, True)
-        indexes.append(primary_key)
+        if primary_key not in indexes:
+            indexes.append(primary_key)
```

Comment:
primary key and indexes might be given as strings or lists, so this won't catch the case of the same index being specified in different ways. Maybe move this fix below after the call to `datastore_helpers.get_list(index)`